pkg/common/resources/resources.go

```diff
@@ -884,6 +918,29 @@ func ComponentWiseMinPermissive(left, right *Resource) *Resource {
 	return out
 }
 
+// ComponentWiseMinPermissiveWithPrecedence Returns a new Resource by giving higher precedence for resource type values present in left when
+// it is present at right as well. Resource type not present in left and present in right would be included too.
+// If either Resource passed in is nil the other Resource is returned
+// If a Resource type is missing from one of the Resource, it is considered empty and the quantity from the other Resource is returned
+func ComponentWiseMinPermissiveWithPrecedence(left, right *Resource) *Resource {
```

Comment:
not the right name we need something better.
At the moment I do not see this covering the action it does